adagrams/game.py

```diff
@@ -1,11 +1,131 @@
+import random
+import array
+
+# manually typed list of all the letters (and how many of them as repeats) that are allowed
+LETTER_LIST = ['A', 'A', 'A', 'A', 'A', 'A', 'A', 'A', 'A', 'B', 'B', 'C', 'C', 'D', 'D', 'D', 'D', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'F', 'F', 'G', 'G', 'G', 'H', 'H', 'I', 'I', 'I', 'I', 'I', 'I', 'I', 'I', 'I', 'J', 'K', 'L', 'L', 'L', 'L', 'M', 'M', 'N', 'N', 'N', 'N', 'N', 'N', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'P', 'P', 'Q', 'R', 'R', 'R', 'R', 'R', 'R', 'S', 'S', 'S', 'S', 'T', 'T', 'T', 'T', 'T', 'T', 'U', 'U', 'U', 'U', 'V', 'V', 'W', 'W', 'X', 'Y', 'Y', 'Z']
+
 def draw_letters():
-    pass
+    letters = [] # empty 'letters' list 
+
+    letters = random.sample(LETTER_LIST,10) #10 letters are randomly selected and set to the letters list
+    
+    return letters #returns 10 randomly selected letters 
 
 def uses_available_letters(word, letter_bank):
-    pass
+
+    #loops through all the letters in the words and makes them lowercase strings in a new list
+    word_case_insensitive =  []
+    for i in word:
+        word_case_insensitive.append(i.lower())
+
+    #loops through all the letters in the letter bank and makes them lowercase strings in a new list
+    letter_bank_case_insensitive = []
+    for i in letter_bank:
+        letter_bank_case_insensitive.append(i.lower())
+
+    #counter for how many loops are completed
+    loop_count = 0
+
+    #for-loop that compares each item in both lists, and removes from letter bank if there's a match
+    for i in word_case_insensitive:
+        for j in letter_bank_case_insensitive:
+            if i == j:
+                loop_count += 1
+                letter_bank_case_insensitive.remove(j)
+
+    #if the loop count and length of a word are equal, it returns True and else False
+    if loop_count == len(word):
+        return True
+    else:
+        return False
+        
+            
+
 
 def score_word(word):
-    pass
+    #lists of scores grouped by number of points. Upper and lower are included to cover all bases (even though it might be not needed since the letters in the word are turned into lowercase)
+    score_1 = ['A', 'a', 'E', 'e', 'I', 'i', 'O', 'o', 'U', 'u', 'L', 'l','N', 'n', 'R', 'r', 'S','s', 'T','t']
+    score_2 = ['D', 'd', 'G', 'g']
+    score_3 = ['B', 'b', 'C', 'c', 'M', 'm','P', 'p']
+    score_4 = ['F', 'f', 'H', 'h', 'V', 'v', 'W', 'w','Y', 'y']
+    score_5 = ['K', 'k']
+    score_8 = ['J','j','X', 'x']
+    score_10 = ['Q', 'q','Z', 'z']
+
+    # Loops through all the letters in the word variable and makes them lowercase strings in a new list
+    word_ =  []
+    for i in word:
+        word_.append(i.lower())
+    
+    #variable is set and intalied to 0 to keep score
+    score = 0
+    #for each letter in 'word_' variable, if its in one of the lists it will add to the score accordingly 
+    for i in word_: 
+        if i in score_1:
```

Comment:
This approach works well but it could be improved by choosing a different data structure. Choosing the best data structure takes practice to get a sense for, but it will come in time! One way to accomplish an easier-to-loop approach is to create a dictionary of letters mapped to point values, e.g. 
```python
letter_values = {
'A': 1,
'B': 3, # and so on
```
then you could use the dictionary to lookup the value in your loop:
```python
    score = 0
    for letter in word.upper():
        score += LETTER_VALUES[letter]
```
Another approach, perhaps more similar to the spirit of yours would be to create a data structure like:
```python
LETTER_VALUES = {
  "AEIOULNRST": 1,
  "DG": 2,
  "BCMP": 3,
  "FHVWY": 4,
  "K": 5,
  "JX": 8,
  "QZ": 10,
}
```
Then you could use two loops to calculate the score:
```python
score = 0
for letter in word.upper():
  for str in LETTER_VALUES:
    if letter in str:
      score += LETTER_VALUES[str]
```
There are many options!